### ISSUE TITLE
[BUG]: exception when field.value is a function

### ISSUE THREAD

**Issue by demsey2** [sensedeep/dynamodb-onetable#46]
in Model.js line 207 we check if field.value exists then we pass to getValueVars where we use regex on the value 
```
// line 222
v.replace(/\${(.*?)}/g, (match, varName)
```
when v is not a string but a function that fails

to fix that I changed line 207 to
```
if (field.value && typeof field.value !== 'function')
```
but I am not sure if that is the right way

**Comment by mobsense** [comment]
Thanks. That will work. Our solution is pushed into getValueVars

```
    getValueVars(v) {
        let list = []
        if (Array.isArray(v)) {
            list = v
        } else if (typeof v != 'function') {
            v.replace(/\${(.*?)}/g, (match, varName) => {
                list.push(varName)
            })
        }
        return list
    }
```

We've added to the documentation the following caveat

>A value function must not depend on the value of other value properties that may or many not have been computed when the function is called.

**Comment by mobsense** [comment]
Fixed in 1.4.5
